src/bobtail/dag.cpp

```diff
@@ -69,9 +69,52 @@ void CBobtailDag::SetId(int16_t new_id)
     id = new_id;
 }
 
+bool CBobtailDag::CheckForCompatibility(CDagNode* newNode)
+{
+    for (auto &tx : newNode->subblock.vtx)
+    {
+        for (auto &input : tx->vin)
+        {
+            if (spent_outputs.count(input.prevout) != 0)
+            {
+                if (spent_outputs[input.prevout] != tx->GetHash())
+                {
+                    // only add it to incompatible_dags if we are not checking
+                    // if the node can be added to this dag
+                    if (id != newNode->dag_id)
```

Comment:
this check sees if the id of the dag we are comparing compat with and the new_node dag_id are the same. When they are the same we are checking if the newnode is compatible with the dag it is attempting to be inserted into. in this scenario we dont want to add the id to the incompatible list because that would say this dag is incompatible with itself which can not happen. 

if the ids are not the same then we are checking if this new node which has been added to the dag with id dag_id breaks compatibility with the current dag we are comparing against. in this scenario we want to add it to the incompatible_dags set so we know that the current dag we are checking against and the dag newNode has been added to are incompatible.